From d33d16c7cb51627c05c82fc5ec0346fcbc6e552c Mon Sep 17 00:00:00 2001
From: Nils Asmussen <nils@os.inf.tu-dresden.de>
Date: Fri, 5 Oct 2012 16:51:06 +0200
Subject: [PATCH 1/2] Ensure that the host-cr2 contains the pf-address for
 SVM-pfs

---
 src/ec_svm.cpp |    1 +
 1 file changed, 1 insertion(+)

diff --git a/src/ec_svm.cpp b/src/ec_svm.cpp
index 1122e23..7e7f79f 100644
--- a/src/ec_svm.cpp
+++ b/src/ec_svm.cpp
@@ -66,6 +66,7 @@ void Ec::svm_exception (mword reason)
                 case Vtlb::GPA_HPA:
                     current->regs.nst_error = 0;
                     current->regs.dst_portal = NUM_VMI - 4;
+                    current->regs.cr2 = cr2;
                     break;
 
                 case Vtlb::GLA_GPA:
-- 
1.7.9.5

