From 225b047582f859932c926b22f52821f03be588bc Mon Sep 17 00:00:00 2001
From: Nils Asmussen <nils@os.inf.tu-dresden.de>
Date: Tue, 2 Oct 2012 11:27:59 +0200
Subject: [PATCH 1/2] No tracing until NOVA is properly initialized

---
 include/stdio.h |   14 +++++++++-----
 1 file changed, 9 insertions(+), 5 deletions(-)

diff --git a/include/stdio.h b/include/stdio.h
index e8721a9..99d960a 100644
--- a/include/stdio.h
+++ b/include/stdio.h
@@ -23,14 +23,18 @@
 #include "console.h"
 #include "cpu.h"
 #include "memory.h"
+#include "hpt.h"
 
 #define trace(T,format,...)                                     \
 do {                                                            \
-    register mword __esp asm ("esp");                           \
-    if (EXPECT_FALSE ((trace_mask & (T)) == (T)))               \
-        Console::print ("[%2ld] " format,                       \
-               ((__esp - 1) & ~PAGE_MASK) == CPU_LOCAL_STCK ?   \
-                Cpu::id : ~0UL, ## __VA_ARGS__);                \
+    Paddr dummy_p;                                              \
+    if (Hptp(Hpt::current()).lookup (CPU_LOCAL_DATA, dummy_p)) {    \
+        register mword __esp asm ("esp");                           \
+        if (EXPECT_FALSE ((trace_mask & (T)) == (T)))               \
+            Console::print ("[%2ld] " format,                       \
+                   ((__esp - 1) & ~PAGE_MASK) == CPU_LOCAL_STCK ?   \
+                   Cpu::id : ~0UL, ## __VA_ARGS__);                \
+    } \
 } while (0)
 
 /*
-- 
1.7.9.5

