From 5e926a9c9d00f9333eba0b1ce78f9de7c7ab9937 Mon Sep 17 00:00:00 2001
From: Nils Asmussen <nils@os.inf.tu-dresden.de>
Date: Thu, 19 Jul 2012 15:26:52 +0200
Subject: [PATCH 3/4] Kill Ec if it causes a pagefault in kernel-area

---
 src/ec_exc.cpp |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ec_exc.cpp b/src/ec_exc.cpp
index d9a324a..1362092 100644
--- a/src/ec_exc.cpp
+++ b/src/ec_exc.cpp
@@ -141,7 +141,7 @@ void Ec::handle_exc (Exc_regs *r)
             break;
     }
 
-    if (r->user())
+    if (r->user() && r->cr2 < USER_ADDR)
         send_msg<ret_user_iret>();
 
     die ("EXC", r);
-- 
1.7.9.5

