From 755b50d8bdf74cdd8036db14b67a8f13a2fb53f2 Mon Sep 17 00:00:00 2001
From: Nils Asmussen <nils@os.inf.tu-dresden.de>
Date: Tue, 2 Oct 2012 11:28:30 +0200
Subject: [PATCH 2/2] Added a util-function to print the backtrace

---
 build/Makefile |    2 +-
 include/util.h |   24 ++++++++++++++++++++++++
 2 files changed, 25 insertions(+), 1 deletion(-)

diff --git a/build/Makefile b/build/Makefile
index f59ff1d..cb0ccb0 100644
--- a/build/Makefile
+++ b/build/Makefile
@@ -62,7 +62,7 @@ $(error $(ARCH) is not a valid architecture)
 endif
 
 # Language options
-FFLAGS		:= -fdata-sections -ffunction-sections -fomit-frame-pointer -freg-struct-return -freorder-blocks -funit-at-a-time
+FFLAGS		:= -fdata-sections -ffunction-sections -freg-struct-return -freorder-blocks -funit-at-a-time
 FFLAGS		+= -fno-asynchronous-unwind-tables -fno-exceptions -fno-rtti
 FFLAGS		+= $(call check,-fno-stack-protector)
 FFLAGS		+= $(call check,-fvisibility-inlines-hidden)
diff --git a/include/util.h b/include/util.h
index 900cc6c..7787ce4 100644
--- a/include/util.h
+++ b/include/util.h
@@ -19,6 +19,10 @@
 #pragma once
 
 #include "compiler.h"
+#include "console.h"
+#include "arch.h"
+#include "memory.h"
+#include "stdio.h"
 
 template <typename T>
 ALWAYS_INLINE
@@ -33,3 +37,23 @@ static inline T max (T v1, T v2)
 {
     return v1 > v2 ? v1 : v2;
 }
+
+static void print_backtrace() {
+	mword bp;
+	asm volatile (
+		"mov	"EXPAND (%%REG(bp))",%0;"
+		: "=a"(bp)
+	);
+	if(bp < CPU_LOCAL_STCK || bp >= CPU_LOCAL_STCK + PAGE_SIZE)
+		return;
+	mword start = bp & ~(PAGE_SIZE - 1);
+	mword end = start + PAGE_SIZE;
+	for(int i = 0; i < 16; i++) {
+		// don't leave that page
+		if(bp < start || bp >= end)
+			break;
+		mword addr = *(reinterpret_cast<mword*>(bp) + 1) - 5;
+		trace(0, "    %p", addr);
+		bp = *reinterpret_cast<mword*>(bp);
+	}
+}
-- 
1.7.9.5

